controllers/operator/deployment.go

```diff
@@ -822,8 +867,31 @@ func generateManagerDeploymentObject(instance *operatorv1alpha1.Authentication,
 	return idpDeployment
 }
 
-func buildIdpVolumes(ldapCACert string, routerCertSecret string) []corev1.Volume {
-	return []corev1.Volume{
+func buildIdpVolumes(ldapCACert string, routerCertSecret string, auditSecretExists bool, required bool, auditSecretName string) []corev1.Volume {
```

Comment:
What is `required` for? Also, providing a Secret name for the TLS Secret would seem to imply that the volume for the audit TLS Secret is needed, so you could just submit the `auditSecretName` and add the volume if it is a non-empty string. Would also require a change ahead of calling this function to be sure that `auditSecretName` is set to `""` if the Secret isn't present. Or you could just get the audit Secret within this code. Whatever you want to do.

Reply:
provider and management need the audit volume, but auth-service does not, I am passing a false for auth-service and true for the other two

Reply:
The way that you could achieve the same effect with the signature I'm suggesting is as follows:

- provider and management: `buildIdpVolumes(ldapCACert, routerCertSecret, auditSecretName)`
- auth service: `buildIdpVolumes(ldapCACert, routerCertSecret, "")`